sanitizer/sanitizer.js

```diff
@@ -1677,11 +1678,84 @@ function handleTagAttr (tag, name, value, isWhiteAttr) {
   }
 }
 
+function handleAttrValue(tag, name, value, cssFilter) {
+  // unescape attribute value firstly
+  value = lib$1.friendlyAttrValue(value);
+
+  if (name === "href" || name === "src") {
+    // filter `href` and `src` attribute
+    // only allow the value that starts with `http://` | `https://` | `mailto:` | `data:` | `/` | `#`
+    value = utilTrim(value);
+    if (value === "#") return "#";
+    if (
+      !(
+        value.substr(0, 7) === "http://" ||
+        value.substr(0, 8) === "https://" ||
+        value.substr(0, 7) === "mailto:" ||
+        value.substr(0, 4) === "tel:" ||
+        value.substr(0, 5) === "data:" ||
+        value[0] === "#" ||
+        value[0] === "/"
```

Comment:
could we maybe do a function here for better readability?

something like

```
function isWhiteListed(value) {
  const whiteList = ['http://', 'https://', 'mailto:', 'tel:', 'data:', '#', '/'];
  return whiteList.some(item => value.indexOf(item) === 0);
}
```

Reply:
This is a copy-paste from XSS, much like `handleTagAttr`. I didn’t see any reason to try to get fancy and deviate from xss logic.

Reply:
I understand, but it does not mean that complexity is not off the chart with that if statement